homeassistant/components/overkiz/climate_entities/atlantic_pass_apc_heating_and_cooling_zone.py

```diff
@@ -0,0 +1,211 @@
+"""Support for Atlantic Pass APC Heating And Cooling Zone Control."""
+from __future__ import annotations
+
+from typing import Any, cast
+
+from pyoverkiz.enums import OverkizCommand, OverkizCommandParam, OverkizState
+
+from homeassistant.components.climate import (
+    PRESET_AWAY,
+    PRESET_COMFORT,
+    PRESET_ECO,
+    PRESET_HOME,
+    PRESET_SLEEP,
+    ClimateEntity,
+    ClimateEntityFeature,
+    HVACMode,
+)
+from homeassistant.const import ATTR_TEMPERATURE, TEMP_CELSIUS
+
+from ..coordinator import OverkizDataUpdateCoordinator
+from ..entity import OverkizEntity
+
+OVERKIZ_TO_HVAC_MODE: dict[str, str] = {
+    OverkizCommandParam.AUTO: HVACMode.AUTO,
+    OverkizCommandParam.ECO: HVACMode.AUTO,
+    OverkizCommandParam.MANU: HVACMode.HEAT,
+    OverkizCommandParam.HEATING: HVACMode.HEAT,
+    OverkizCommandParam.STOP: HVACMode.OFF,
+    OverkizCommandParam.INTERNAL_SCHEDULING: HVACMode.AUTO,
+    OverkizCommandParam.COMFORT: HVACMode.HEAT,
+}
+
+HVAC_MODE_TO_OVERKIZ = {v: k for k, v in OVERKIZ_TO_HVAC_MODE.items()}
+
+OVERKIZ_TO_PRESET_MODES: dict[str, str] = {
+    OverkizCommandParam.OFF: PRESET_ECO,
+    OverkizCommandParam.STOP: PRESET_ECO,
+    OverkizCommandParam.MANU: PRESET_COMFORT,
+    OverkizCommandParam.COMFORT: PRESET_COMFORT,
+    OverkizCommandParam.ABSENCE: PRESET_AWAY,
+    OverkizCommandParam.ECO: PRESET_ECO,
+    OverkizCommandParam.INTERNAL_SCHEDULING: PRESET_HOME,
+}
+
+PRESET_MODES_TO_OVERKIZ = {v: k for k, v in OVERKIZ_TO_PRESET_MODES.items()}
+
+OVERKIZ_TO_PROFILE_MODES: dict[str, str] = {
+    OverkizCommandParam.OFF: PRESET_SLEEP,
+    OverkizCommandParam.STOP: PRESET_SLEEP,
+    OverkizCommandParam.ECO: PRESET_ECO,
+    OverkizCommandParam.ABSENCE: PRESET_AWAY,
+    OverkizCommandParam.MANU: PRESET_COMFORT,
+    OverkizCommandParam.DEROGATION: PRESET_COMFORT,
+    OverkizCommandParam.COMFORT: PRESET_COMFORT,
+}
+
+
+class AtlanticPassAPCHeatingAndCoolingZone(OverkizEntity, ClimateEntity):
+    """Representation of Atlantic Pass APC Heating And Cooling Zone Zone Control."""
+
+    _attr_hvac_modes = [*HVAC_MODE_TO_OVERKIZ]
+    _attr_preset_modes = [*PRESET_MODES_TO_OVERKIZ]
+    _attr_supported_features = (
+        ClimateEntityFeature.TARGET_TEMPERATURE | ClimateEntityFeature.PRESET_MODE
+    )
+    _attr_temperature_unit = TEMP_CELSIUS
+
+    def __init__(
+        self, device_url: str, coordinator: OverkizDataUpdateCoordinator
+    ) -> None:
+        """Init method."""
+        super().__init__(device_url, coordinator)
+        self.temperature_device = self.executor.linked_device(
+            int(self.device_url.split("#", 1)[1]) + 1
```

Comment:
Could you elaborate what is happening here? Is the temperature device that you need always the current device + 1? Perhaps we can add a comment here, since the `1, [1] +1` looks a bit confusing :-).

` device URL (<protocol>://<gatewayId>/<deviceAddress>[#<subsystemId>])`
Perhaps we should add a function that can split the device url and return `gateway_id`, `device_address` and `subsystem_id`.

Reply:
Added a `self.index_device_url` in entity.py when existing.  
Update the commented code to explain it in a comment a simplify the +1